crates/loro-internal/src/undo.rs

```diff
@@ -76,7 +76,23 @@ pub struct UndoManager {
     inner: Arc<Mutex<UndoManagerInner>>,
 }
 
-#[derive(Debug)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum UndoOrRedo {
+    Undo,
+    Redo,
+}
+impl UndoOrRedo {
+    fn opposite(&self) -> UndoOrRedo {
+        match self {
+            Self::Undo => Self::Redo,
+            Self::Redo => Self::Undo,
+        }
+    }
+}
+
+pub type OnPush = Box<dyn Fn(UndoOrRedo, CounterSpan, &mut LoroValue) + Send + Sync>;
```

Comment:
Yeah, it could be hard to understand. But it's useful in certain cases. For example,, you can infer the length of the ops involved in this undo/redo. And you can use it to query the related ops.